src/k8s/pkg/client/k8sd/interface.go

```diff
@@ -35,6 +35,14 @@ type ConfigClient interface {
 	SetClusterConfig(context.Context, apiv1.SetClusterConfigRequest) error
 }
 
+// ManagementClient implements methods to manage the cluster.
+type ManagementClient interface {
```

Comment:
Hm, I feel like Management is a bit too generic. 

Maybe `Lifecycle` or `Maintainance`? What things do you plan to cover with this client apart from certificate refreshes?

Reply:
Honestly, the certificate refresh functionality didn’t fit with any of the other clients. ClusterClient is quite microcluster-centric, and the other clients are too specific to accommodate the certificate refresh endpoints. I can change this to the Maintenance client instead, but if you have a client that should adopt these endpoints, please let me know.

Reply:
yy, this was just me thinking about a good name for that client. I'm not good at naming things, so don't take my suggestions as the only way forward.
My point was that `ManagementClient` is a bit too generic and we start stuffing everything in there. 

You could either be super specific (`CertificateRefreshClient`) or make this a bit more specific then `Management` (not sure if `Lifecycle` or `Maintainance` is any better, though).

Reply:
Let's merge both and use `ClusterMaintenanceClient`. I think it should better reflect the purpose of the client.

Reply:
sounds good